profiling/src/profile/pprof/sliced_proto.rs

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+//! This file is a companion to the standard pprof protobuf definition.
+//! `Profile` fields are "sliced" into separate messages, allowing them to be
+//! serialized in a streamed manner.  This takes advantage of the fact that
+//! the top level message in a protobuf doesn't have a length header, so the
+//! bits on the wire are indistinguishable between serializing these sliced
+//! messages, and serializing the top-level message.
```

Comment:
May be worth adding here, for future context, why streamed emission is useful ;)

Reply:
done